cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java

```diff
@@ -261,13 +265,18 @@ protected static <C extends KafkaConnectCluster> C fromSpec(Reconciliation recon
         result.gcLoggingEnabled = spec.getJvmOptions() == null ? JvmOptions.DEFAULT_GC_LOGGING_ENABLED : spec.getJvmOptions().isGcLoggingEnabled();
 
         result.jvmOptions = spec.getJvmOptions();
-        result.metrics = new MetricsModel(spec);
+
+        if (spec.getMetricsConfig() instanceof JmxPrometheusExporterMetrics) {
+            result.metrics = new JmxPrometheusExporterModel(spec);
+        } else if (spec.getMetricsConfig() instanceof StrimziMetricsReporter) {
+            LOGGER.errorCr(reconciliation, "The Strimzi Metrics Reporter is not supported with this component");
+            throw new InvalidResourceException("The Strimzi Metrics Reporter is not supported with this component");
```

Comment:
From Cruise cOntrol you seemed to have added CEL validation for this. Should you do it for Connect / MM2 as well?

Reply:
Answered above.

Reply:
CEL validation and if else now present in CC KafkaConnect and MM2.

Reply:
That's a question I also asked before, do we need this check even with the CEL validation in place?